### ISSUE TITLE
Test run fails on OSX with locked database

### ISSUE THREAD

**Issue by tracywalker** [mozilla/iris#253]
Tried to run a single test and got this report:

-------------------------------------------------Test Report--------------------------------------------------
Application: Firefox, Platform: osx, fx_version: 67.0, fx_build_id: 20190509185224, channel: beta
Passed: 0, Failed: 1, Skipped: 0, Errors 0  -- Total: 1               Total time: 19.621957778930664 second(s)

The following tests did not pass:
open_url_in_tab.py

--------------------------------------------------------------------------------------------------------------

----------------------------------------Test session firefox complete-----------------------------------------
Terminating process.
Exception ignored in: <function BaseProfile.__del__ at 0x104a541e0>
Traceback (most recent call last):
  File "/Users/tracy/.local/share/virtualenvs/iris2-AhcqVXCe/lib/python3.7/site-packages/mozprofile/profile.py", line 76, in __del__
  File "/Users/tracy/.local/share/virtualenvs/iris2-AhcqVXCe/lib/python3.7/site-packages/mozprofile/profile.py", line 262, in cleanup
  File "/Users/tracy/.local/share/virtualenvs/iris2-AhcqVXCe/lib/python3.7/site-packages/mozprofile/permissions.py", line 417, in clean_db
sqlite3.OperationalError: database is locked

**Comment by mwobensmith** [comment]
@mariusandreifilip I think this might be the issue we discussed where Iris stalls/fails at the end of a run.

**Comment by tracywalker** [comment]
The test itself never started.  The test run downloaded the build and launched Firefox with the Iris logo ion content  (no test case and in progress meter),  then stalled out instead of proceeding with the test.

**Comment by mwobensmith** [comment]
The progress bar is just missing - I haven't finished that yet.

For me, the test launches the browser and executes correctly. When it completes, we get to the "Terminating process" part in the terminal and it just hangs. 

I can definitely reproduce with this test, so that's hopeful.

**Comment by mwobensmith** [comment]
This also reproduces the error:

`iris firefox -t save_image_controls`

**Comment by mariusandreifilip** [comment]
Indeed the problem reproduces when we try to change prefs  and it's a bug in the Mozprofile library .

def clean_db(self):
        """Removed permissions added by mozprofile."""

        sqlite_file = os.path.join(self._profileDir, "permissions.sqlite")
        if not os.path.exists(sqlite_file):
            return

        # Open database and create table
        permDB = sqlite3.connect(sqlite_file)
        cursor = permDB.cursor()

        # TODO: only delete values that we add, this would require sending
        # in the full permissions object
        cursor.execute("DROP TABLE IF EXISTS moz_hosts")

        # Commit and close
        permDB.commit()
        cursor.close() 

I will try to make a workaround to this issue

**Comment by mariusandreifilip** [comment]
fixed by issue_253 and merged in dev.Tested on 2 macs and a non retina ans seems to be fixed

**Comment by tracywalker** [comment]
Still fails as originally reported on dev branch for a few minutes ago.

----------------------------------------Test session firefox complete-----------------------------------------
Terminating process.
Exception ignored in: <function BaseProfile.__del__ at 0x103be3048>
Traceback (most recent call last):
  File "/Users/tracy/.local/share/virtualenvs/iris2-AhcqVXCe/lib/python3.7/site-packages/mozprofile/profile.py", line 76, in __del__
  File "/Users/tracy/.local/share/virtualenvs/iris2-AhcqVXCe/lib/python3.7/site-packages/mozprofile/profile.py", line 262, in cleanup
  File "/Users/tracy/.local/share/virtualenvs/iris2-AhcqVXCe/lib/python3.7/site-packages/mozprofile/permissions.py", line 417, in clean_db
sqlite3.OperationalError: database is locked

**Comment by mwobensmith** [comment]
Still fails for me too:

```
-----------------------------------------------Test session firefox complete-----------------------------------------------
Terminating process.
Exception ignored in: <function BaseProfile.__del__ at 0x106b257b8>
Traceback (most recent call last):
  File "/Users/mwobensmith/.local/share/virtualenvs/iris2-Fh8pD0FE/lib/python3.7/site-packages/mozprofile/profile.py", line 76, in __del__
    self.cleanup()
  File "/Users/mwobensmith/.local/share/virtualenvs/iris2-Fh8pD0FE/lib/python3.7/site-packages/mozprofile/profile.py", line 262, in cleanup
    self.permissions.clean_db()
  File "/Users/mwobensmith/.local/share/virtualenvs/iris2-Fh8pD0FE/lib/python3.7/site-packages/mozprofile/permissions.py", line 417, in clean_db
    cursor.execute("DROP TABLE IF EXISTS moz_hosts")
sqlite3.OperationalError: database is locked
```

**Comment by tracywalker** [comment]
I believe Matt and I are both on Mojave (OSX 10.14.5)  Is anyone who doesn't see this failure also on Mojave?

**Comment by mwobensmith** [comment]
@mariusandreifilip This still fails with the latest fixes. Can you try again with this test?

`iris firefox -t save_image_controls`

**Comment by mariusandreifilip** [comment]
strange that is still reproducible ,I will try it

**Comment by tracywalker** [comment]
I think the locked database issue is solved for me.  Now tests don't run as described in https://github.com/mozilla/iris2/issues/397.

**Comment by mwobensmith** [comment]
I see this behavior occur when a browser is left open from a previous test, due to inability to force quit while a modal dialog is open. That part has been fixed in #425, which might make this issue go away. Keeping open for now, will close if we don't see it within the next few weeks.
